python/cudf/cudf/_lib/cudf.pyx

```diff
@@ -176,12 +176,13 @@ cdef gdf_scalar* gdf_scalar_from_scalar(val, dtype=None) except? NULL:
     cdef gdf_scalar* s = <gdf_scalar*>malloc(sizeof(gdf_scalar))
     if s is NULL:
         raise MemoryError
+    if val is None:
+        is_valid = False
+        val = dtype.type(0)
```

Comment:
Future optimization note: would be nice if we didn't have to initialize this for `None` / `null` values.